code/modules/reagents/chemistry/reagents.dm

```diff
@@ -39,12 +41,16 @@
 /datum/reagent/proc/reaction_mob(mob/living/M, method=TOUCH, reac_volume, show_message = 1, touch_protection = 0)
 	if(!istype(M))
 		return 0
-	if(method == VAPOR) //smoke, foam, spray
-		if(M.reagents)
-			var/modifier = Clamp((1 - touch_protection), 0, 1)
-			var/amount = round(reac_volume*modifier, 0.1)
-			if(amount >= 0.5)
-				M.reagents.add_reagent(id, amount)
+	if(method == VAPOR || TOUCH) //smoke, foam, spray
```

Comment:
group your conditionals

Reply:
what that mean

Reply:
Instead of:
If(x)
   If(y)
      If(z)
         Do(thing)

Group them like this:
If(x & y & z)
   Do(thing)